### PR TITLE
feat: cap level to element size in shaka + init option

### PR DESCRIPTION
**Pull request by slowmove**

### CODE REVIEW

File: packages/core/src/WebPlayer.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ export { IPlayerState, IVideoLevel } from './tech/BaseTech';
 
 export interface IWebPlayerOptions {
   video: HTMLVideoElement;
+  capLevelToElementSize?: boolean;
```
Comment (slowmove) [inline packages/core/src/WebPlayer.ts:12]: New init option

File: packages/core/src/tech/HlsJsTech.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
   PlaybackState,
 } from './BaseTech';
 import Hls from 'hls.js';
 import { PlayerEvent } from '../util/constants';
+import { IWebPlayerOptions } from '../WebPlayer';
 
 const DEFAULT_CONFIG = {
   capLevelOnFPSDrop: true,
-  capLevelToPlayerSize: true,
+  capLevelToPlayerSize: false,
```
Comment (slowmove) [inline packages/core/src/tech/HlsJsTech.ts:11]: Changed default behaviour - might be something to discuss?

Comment (birme) [reply packages/core/src/tech/HlsJsTech.ts:11]: Just thinking if the previous default is relevant to when used as web-component?

Comment (slowmove) [reply packages/core/src/tech/HlsJsTech.ts:11]: I mean from a bandwidth optimization it is always relevant to not play out higher resolution than what can be displayed due to the element's pixel size.

Maybe change it the other way around then - `disablePlayerSizeLevelCap`?

Comment (birme) [reply packages/core/src/tech/HlsJsTech.ts:11]: Yes, sounds great!

Comment (birme) [reply packages/core/src/tech/HlsJsTech.ts:11]: Perhaps expose that option in the web-component while at it?

Comment (slowmove) [reply packages/core/src/tech/HlsJsTech.ts:11]: Absolutely - and I'll update the Readme as well.

**Review by birme — APPROVED** [review]

**Pull request merged**